### PR TITLE
E-mail Invalid Handler

### PR DESCRIPTION
**Pull request by MTS11648**
Will ask the customer for e-mail verification if the e-mail does not contain common e-mail keywords like "at". [new and improved version]

### CODE REVIEW

File: AutoBotCSharp/Agent.cs [hunk 1]
```diff
@@ -1125,6 +1126,12 @@ public static string CleanUpResponse(string resp)
                                 Conversation.RollTheClip(clip);
                                 break;
                             }
+                            else if (!response.Contains("at") && !response.Contains("@") && !response.Contains("mail") && !response.Contains("com") && !EmailInvalid_Stated)
+                            {
+                                askForRepeat();
+                                EmailInvalid_Stated = true;
```
Comment (SpadesNeil) [inline AutoBotCSharp/Agent.cs]: I think this variable needs to be reset during CALL_START so it doesn't get mistakenly carried over to a new call.

On line 839, add ``EmailInvalid_Stated = false;``

File: AutoBotCSharp/Agent.cs [hunk 2]
```diff
@@ -835,6 +836,7 @@ public static string CleanUpResponse(string resp)
                 {
                     switch (AutoFormQuestion)
                     {
+                        EmailInvalid_Stated = false;
                         case Agent.CALL_START:
                             CurrentCustomer.NoEmail_Count = 0;
```
Comment (SpadesNeil) [inline AutoBotCSharp/Agent.cs]: Sorry, I was referring to this line. I'll just fix it.

**Comment by SpadesNeil** [comment]
Just needs proper testing now.

**Comment by SpadesNeil** [comment]
In future we need to test better. :P

**Comment by MTS11648** [comment]
@SpadesNeil funny because it worked in some calls, but not all of them. it didn't break anything, don't see why we're reverting it, but okie

**Comment by SpadesNeil** [comment]
Because in the calls it doesn't work in, Cheryl just interrupts the email question by studdering and then saying "I just need an email that you do have access to" again. This a the sign of problems.

**Comment by MTS11648** [comment]
oh I haven't seen that type of behavior yet; not sure why it would do that, i guess i'll have to look into that when i get a free chance.

**Pull request merged**